### PR TITLE
Adding QwiicLEDStrip

### PR DESCRIPTION
**Pull request by alan412**
The QwiicLED Strip is from Sparkfun - https://www.sparkfun.com/products/14783
 Driver donated by FTC16072 - Quantum Quacks

### CODE REVIEW

**Comment by ItsHarper** [comment]
@Windwoes Reminder about this PR

**Comment by lizlooney** [comment]
@Windwoes I might be interested in this as an external library.

**Comment by Windwoes** [comment]
Whoops, I actually never saw this initially because I somehow never changed my settings to watch this repo...

File: i2cdrivers/build.gradle [hunk 1]
```diff
@@ -35,6 +35,7 @@ android {
 dependencies {
     //compileOnly fileTree(include: ['*.aar'], dir: '../libs')
     compileOnly 'org.openftc:robotcore-stock:5.1'
+    implementation 'com.android.support:support-compat:28.0.0'
```
Comment (Windwoes) [inline i2cdrivers/build.gradle]: The new SDK versions use androidx, so having appcompat as a dependency here will cause a compile time conflict. Can you change this to `compileOnly 'androidx.appcompat:appcompat:1.2.0'`?

You can then use this import: `import androidx.annotation.ColorInt;`

File: i2cdrivers/src/main/java/org/openftc/i2cdrivers/QwiicLEDStrip.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+import com.qualcomm.robotcore.hardware.I2cAddr;
+import com.qualcomm.robotcore.hardware.I2cDeviceSynchDevice;
+import com.qualcomm.robotcore.hardware.I2cDeviceSynchSimple;
+import com.qualcomm.robotcore.hardware.I2cWaitControl;
+import com.qualcomm.robotcore.hardware.configuration.annotations.DeviceProperties;
+import com.qualcomm.robotcore.hardware.configuration.annotations.I2cDeviceType;
+
+@I2cDeviceType()
+@DeviceProperties(name = "QWIIC LED Strip", description = "Sparkfun QWIIC LED Strip", xmlTag = "QWIIC_LED_STRIP")
```
Comment (Windwoes) [inline i2cdrivers/src/main/java/org/openftc/i2cdrivers/QwiicLEDStrip.java]: Is this actually called an LED Strip? I can only find reference to a Qwiic LED "stick" online.

File: i2cdrivers/src/main/java/org/openftc/i2cdrivers/QwiicLEDStrip.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+
+    /**
+     * Change the color of a specific LED
+     *
+     * @param position which LED to change (1 - 255)
+     * @param color    what color to set it to
+     */
+    public void setColor(int position, @ColorInt int color) {
+        byte[] data = new byte[4];
+        data[0] = (byte) position;
```
Comment (Windwoes) [inline i2cdrivers/src/main/java/org/openftc/i2cdrivers/QwiicLEDStrip.java]: Assuming this is for the [Qwicc LED **Stick**](https://www.sparkfun.com/products/14783), I think the Javadoc comment is incorrect about the range for `position`, shouldn't it be 0-9?

Comment (Windwoes) [reply i2cdrivers/src/main/java/org/openftc/i2cdrivers/QwiicLEDStrip.java]: Looking pictures of the PCB, I see the `DO` and `CO` lines are exposed on the right side, is the idea that you would add an extra APA102 strip (like an Adafruit Dotstar) to those lines to continue the strip?

File: i2cdrivers/build.gradle (hunk 1 above)
Comment (alan412) [reply i2cdrivers/build.gradle]: I'll make this change.   Good call.

File: i2cdrivers/src/main/java/org/openftc/i2cdrivers/QwiicLEDStrip.java (hunk 2 above)
Comment (alan412) [reply i2cdrivers/src/main/java/org/openftc/i2cdrivers/QwiicLEDStrip.java]: you are right.   It is the LED Stick.  I'll make that change

File: i2cdrivers/src/main/java/org/openftc/i2cdrivers/QwiicLEDStrip.java (hunk 3 above)
Comment (alan412) [reply i2cdrivers/src/main/java/org/openftc/i2cdrivers/QwiicLEDStrip.java]: That is exactly right.   That is why it can be up to 255.

**Comment by Windwoes** [comment]
Many apologies for letting this sit unmerged for so long; now that I've got maven central set up after the bintray shutdown, I think I can do a first release of this library. Is this PR still ready to go?

**Comment by alan412** [comment]
I don't see any reason it needs to change.   I tried it with the FTC
release of 6.2 to make sure it still works.

--Alan

**Review by ItsHarper — APPROVED** [review]
This looks like a nice reasonable driver, nice job.

**Comment by Windwoes** [comment]
Cool, I'll see about getting a first release out in the next few days.

**Comment by Windwoes** [comment]
v1.0.0 released :)

**Pull request merged**